File: docs/deploy.py
from markdown import markdown
import os

# load head, midd, foot

head_file = open('html/head.html')
head = head_file.read()
head_file.close()

midd_file = open('html/midd.html')
midd = midd_file.read()
midd_file.close()

foot_file = open('html/foot.html')
foot = foot_file.read()
foot_file.close()

mds = os.listdir('md')
for md in mds:
	
	# convert md to html
	body_file = open('md/' + md)
	body = markdown(body_file.read())
	body_file.close()

	title = md[:-3]

	# combine everything, write to file
	site_file = open(title + '.html', 'w')
	site_file.write(head + title + midd + body + foot)
	site_file.close()

File: docs/deploy.go
package main

import (
	"os/exec"
	"io/ioutil"
	"strings"
	"github.com/russross/blackfriday"
)

func main () {

	// list of md files
	out, _ := exec.Command("ls", "md").Output()
	mds := strings.Fields(string(out))

	// for each, generate site file
	for _, md := range mds {

		// title
		temp := md[:len(md) - 3]
		ioutil.WriteFile("title.temp", []byte(temp), 0666)

		// body
		in, _ := ioutil.ReadFile("md/" + md)
		out = blackfriday.MarkdownBasic(in)
		ioutil.WriteFile("body.temp", out, 0666)

		// cat everything
		out, _ = exec.Command("cat", "html/head.html", "title.temp", "html/midd.html", "body.temp", "html/foot.html").Output()
		ioutil.WriteFile(temp + ".html", out, 0666)
	}

	// cleanup
	exec.Command("rm", "title.temp", "body.temp").Run()
}
